var/spack/repos/builtin/packages/emboss/package.py

```diff
@@ -16,9 +16,20 @@ class Emboss(AutotoolsPackage):
 
     version('6.6.0', 'cc3fca80cb0618deb10fa0d29fe90e4b')
 
-    depends_on('libxpm')
+    variant('X', default=True, description='Build with X support')
+    variant('postgresql', default=True, description='Build with postgreSQL support')
+
+    depends_on('libxpm', when='+X')
     depends_on('libgd')
-    depends_on('postgresql')
+    depends_on('postgresql', when='+postgresql')
+
+    def configure_args(self):
+        args = []
+        if self.spec.satisfies('~X'):
```

Comment:
I was thinking it might be preferable for my PR to result in no changes for people building without variant flags (i.e., if one did a `spack install emboss` after vs. before my PR, the configure and build routine would be the same).  But, accept the voice of experience re: bites.

Reply:
Please let me know if the changes in 88457b2 look reasonable now.  (Github's new "resolve" button will take some getting used to but I think I like it).